File: src/android/input/InitInput.java
package com.bandyer.cordova.plugin.input;

import org.json.JSONArray;
import org.json.JSONObject;

import com.bandyer.cordova.plugin.Constants;

public class InitInput {

    private String mAppId;
    private String mEnvironment;
    private boolean mIsCallEnabled;
    private boolean mIsFileSharingEnabled;
    private boolean mIsWhiteboardEnabled;
    private boolean mIsChatEnabled;
    private boolean mIsLogEnabled;

    public static InitInput createFrom(JSONArray argsArray) throws PluginInputNotValidException {
        InitInput initInput = new InitInput();
        try {
            JSONObject args = (JSONObject) argsArray.get(0);
            String appId = args.getString(Constants.ARG_APP_ID);
            if(appId == null || appId.equals("")) {
                throw new PluginInputNotValidException(Constants.ARG_APP_ID + " cannot be null");
            }
            initInput.setAppId(appId);
            initInput.setCallEnabled(args.getBoolean(Constants.ARG_CALL_ENABLED));
            initInput.setChatEnabled(args.getBoolean(Constants.ARG_CHAT_ENABLED));
            initInput.setEnvironment(args.getString(Constants.ARG_ENVIRONMENT));
            initInput.setFileSharingEnabled(args.getBoolean(Constants.ARG_FILE_SHARING_ENABLED));
            initInput.setWhiteboardEnabled(args.getBoolean(Constants.ARG_WHITEBOARD_ENABLED));
            initInput.setWhiteboardEnabled(args.getBoolean(Constants.ARG_WHITEBOARD_ENABLED));
            initInput.setLogEnabled(args.getBoolean(Constants.ARG_ENABLE_LOG));
            return initInput;
        }catch (Throwable t) {
            throw new PluginInputNotValidException("error on InitInput " + t.getMessage(), t);
        }

    }

    public String getAppId() {
        return mAppId;
    }

    public void setAppId(String appId) {
        mAppId = appId;
    }

    public String getEnvironment() {
        return mEnvironment;
    }

    public void setEnvironment(String environment) {
        mEnvironment = environment;
    }

    public boolean isCallEnabled() {
        return mIsCallEnabled;
    }

    public void setCallEnabled(boolean callEnabled) {
        mIsCallEnabled = callEnabled;
    }

    public boolean isFileSharingEnabled() {
        return mIsFileSharingEnabled;
    }

    public void setFileSharingEnabled(boolean fileSharingEnabled) {
        mIsFileSharingEnabled = fileSharingEnabled;
    }

    public boolean isWhiteboardEnabled() {
        return mIsWhiteboardEnabled;
    }

    public void setWhiteboardEnabled(boolean whiteboardEnabled) {
        mIsWhiteboardEnabled = whiteboardEnabled;
    }

    public boolean isChatEnabled() {
        return mIsChatEnabled;
    }

    public void setChatEnabled(boolean chatEnabled) {
        mIsChatEnabled = chatEnabled;
    }

    public boolean isProdEnvironment(){
        return Constants.VALUE_ENVIRONMENT_PRODUCTION.equals(getEnvironment());
    }

    public boolean isSandoboxEnvironment(){
        return Constants.VALUE_ENVIRONMENT_SANDBOX.equals(getEnvironment());
    }

    public boolean isLogEnabled() {
        return mIsLogEnabled;
    }

    public void setLogEnabled(boolean logEnabled) {
        mIsLogEnabled = logEnabled;
    }
}

File: src/android/src/main/java/com/bandyer/cordova/plugin/input/InitInput.kt
package com.bandyer.cordova.plugin.input

import org.json.JSONArray
import org.json.JSONObject
import android.util.Log

import com.bandyer.cordova.plugin.Constants

class InitInput {

    var appId: String? = null
    var environment: String? = null
    var isCallEnabled: Boolean = false
    var isFileSharingEnabled: Boolean = false
    var isScreenSharingEnabled: Boolean = false
    var isWhiteboardEnabled: Boolean = false
    var isChatEnabled: Boolean = false
    var isLogEnabled: Boolean = false


    val isProdEnvironment: Boolean
        get() = Constants.VALUE_ENVIRONMENT_PRODUCTION == environment

    val isSandoboxEnvironment: Boolean
        get() = Constants.VALUE_ENVIRONMENT_SANDBOX == environment

    companion object {

        @Throws(PluginInputNotValidException::class)
        fun createFrom(argsArray: JSONArray): InitInput {
            val initInput = InitInput()
            try {
                val args = argsArray.get(0) as JSONObject
                val appId = args.getString(Constants.ARG_APP_ID)
                if (appId == null || appId == "") {
                    throw PluginInputNotValidException(Constants.ARG_APP_ID + " cannot be null")
                }
                initInput.appId = appId
                initInput.isCallEnabled = args.getBoolean(Constants.ARG_CALL_ENABLED)
                initInput.isChatEnabled = args.getBoolean(Constants.ARG_CHAT_ENABLED)
                initInput.environment = args.getString(Constants.ARG_ENVIRONMENT)
                initInput.isFileSharingEnabled = args.getBoolean(Constants.ARG_FILE_SHARING_ENABLED)
                initInput.isWhiteboardEnabled = args.getBoolean(Constants.ARG_WHITEBOARD_ENABLED)
                initInput.isScreenSharingEnabled = args.getBoolean(Constants.ARG_SCREENSHARING_ENABLED)



                initInput.isLogEnabled = args.getBoolean(Constants.ARG_ENABLE_LOG)
                return initInput
            } catch (t: Throwable) {
                throw PluginInputNotValidException("error on InitInput " + t.message, t)
            }

        }
    }
}
